pelican/tests/test_settings.py

```diff
@@ -182,4 +183,73 @@ def test_theme_and_extra_templates_exception(self):
         settings['EXTRA_TEMPLATES_PATHS'] = ['/ha']
         settings['THEME_TEMPLATES_OVERRIDES'] = ['/foo/bar']
 
-        self.assertRaises(Exception, configure_settings, settings)
+        self.assertRaises(Exception, handle_deprecated_settings, settings)
+
+    def test_slug_and_slug_regex_substitutions_exception(self):
+        settings = {}
+        settings['SLUG_REGEX_SUBSTITUTIONS'] = [('C++', 'cpp')]
+        settings['TAG_SUBSTITUTIONS'] = [('C#', 'csharp')]
+
+        self.assertRaises(Exception, handle_deprecated_settings, settings)
+
+    def test_deprecated_slug_substitutions(self):
+        default_slug_regex_subs = self.settings['SLUG_REGEX_SUBSTITUTIONS']
+        settings = {}
+        settings = handle_deprecated_settings(settings)
+        self.assertNotIn('SLUG_REGEX_SUBSTITUTIONS', settings)
+        self.assertNotIn('TAG_REGEX_SUBSTITUTIONS', settings)
+        self.assertNotIn('CATEGORY_REGEX_SUBSTITUTIONS', settings)
+        self.assertNotIn('AUTHOR_REGEX_SUBSTITUTIONS', settings)
```

Comment:
Not sure what's tested here. "`handle_deprecated_settings` doesn't pollute"?

Reply:
Yes, in particular because `DEFAULT_CONFIG` can only take effect if these are not already present in the settings.

Reply:
Fair enough, then in that regard, can I request some comment explaining each "test group" in this function?

Reply:
Something like:
```
# If no deprecated setting was there, don't produce new ones
....
# Transform SLUG_SUBSTITUTIONS, shouldn't create {TAG, CATEGORY}_REGEX_SUBSTITUTIONS
...
# etc
```